src/export_vdf/vertexai_vector_search_export.py

```diff
@@ -0,0 +1,236 @@
+                                index_names.append(d.index)
+
+        except ValueError as ve:
+            print("Could not find given index value")
+            raise ve
+
+        index_metas = {}
+        for index_name in tqdm(index_names, desc="Fetching indexes"):
+            index_meta = self.get_data_for_index(index_name)
+            index_metas[index_name] = index_meta
+
+        # Create and save internal metadata JSON
+        self.file_structure.append(os.path.join(self.vdf_directory, "VDF_META.json"))
+        internal_metadata = {
+            "version": self.args["library_version"],
+            "file_structure": self.file_structure,
+            "author": os.environ.get("USER"),
+            "exported_from": self.DB_NAME_SLUG,
+            "indexes": index_metas,
+        }
+        with open(os.path.join(self.vdf_directory, "VDF_META.json"), "w") as json_file:
+            json.dump(internal_metadata, json_file, indent=4)
+        # print internal metadata properly
+        tqdm.write(json.dumps(internal_metadata, indent=4))
+        return True
+
+    def get_index_endpoint_name(self, index_name):
+        index = MatchingEngineIndex(index_name=index_name)
+        index_endpoints = [
+            (deployed_index.index_endpoint, deployed_index.deployed_index_id)
+            for deployed_index in index.deployed_indexes
+        ]
+
+        index_endpoint_name = None
+        if len(index_endpoints) > 0:
+            index_endpoint_name = index_endpoints[0][0]
+            deployed_index_id = index_endpoints[0][1]
+        else:
+            raise Exception(
+                "Index not deployed to an endpoint. Cannot export index data"
+            )
+
+        return index_endpoint_name, deployed_index_id
+
+    def get_data_for_index(self, index_name):
+        index_meta_list = []
+        # get index endpoint resource id and deployed index id
+        index_endpoint_name, deployed_index_id = self.get_index_endpoint_name(
+            index_name=index_name
+        )
+        # define index and index endpoint
+        index = MatchingEngineIndex(index_name=index_name)
+        index_endpoint = MatchingEngineIndexEndpoint(index_endpoint_name)
+
+        vectors_directory = os.path.join(self.vdf_directory, index.display_name)
+        os.makedirs(vectors_directory, exist_ok=True)
+
+        # get index metadata
+        index_meta = index.to_dict()
+        total = int(index_meta.get("indexStats", {}).get("vectorsCount"))
+        if self.max_vectors:
+            total = self.max_vectors
+
+        dim = int(index_meta.get("metadata", {}).get("config", {}).get("dimensions", 0))
+
+        # start exporting
+        pbar = tqdm(total=total, desc=f"Exporting {index.display_name}")
+        # find nearest neighbors as proxy to export all datapoint ids
+        neighbors = index_endpoint.find_neighbors(
+            deployed_index_id=deployed_index_id,
+            queries=[[0.0] * dim],
+            num_neighbors=total,
```

Comment:
Is there a max num_neighbors that can be queried. If so, what happens if total is more than that.

Reply:
No hard limit on num_neighbors to query